tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -147,131 +164,79 @@ Note that:
 
 #### Step 3: Create the pPCA model
 
-```julia
-@model function pPCA(x, ::Type{TV}=Array{Float64}) where {TV}
-    # Dimensionality of the problem.
-    N, D = size(x)
-
-    # latent variable z
-    z ~ filldist(Normal(), D, N)
-
-    # side note for the curious
-    # we use the more concise filldist syntax partly for compactness, but also for compatibility with other AD
-    # backends, see the [Turing Performance Tipps](https://turing.ml/dev/docs/using-turing/performancetips)
-    # w = TV{2}(undef, D, D)
-    # for d in 1:D
-    #  w[d, :] ~ MvNormal(ones(D))
-    # end
-
-    # weights/loadings W
-    w ~ filldist(Normal(), D, D)
-
-    # mean offset
-    m ~ MvNormal(ones(D))
-    mu = (w * z .+ m)'
-    for d in 1:D
-        x[:, d] ~ MvNormal(mu[:, d], ones(N))
-    end
-end;
-```
+Here we construct the probabilistic model `pPCA()`. As per the p-PCA formula, we think of each row (i.e. each gene feature) following a $N=60$ dimensional multivariate normal distribution centered around the corresponding row of $\mathbf{W}_{D \times k} \times \mathbf{Z}_{k \times N} + \boldsymbol{\mu}_{D \times N}$.
 
 ```julia
-@model function pPCA(x, k)
-    # function pPCA(k::Int64, x::Type{TV}=Array{Float64}) where {TV}
+@model function pPCA(X::Array{Float64}, k::Int64)
```

Comment:
You basically never want to use `Int64` explicitly. Also `X` should be a `Matrix` (or, more generally, a `AbstractMatrix`) and the type of its elements is not really relevant (could be `<:Real` but `Float64` is quite restrictive):
```suggestion
@model function pPCA(X::AbstractMatrix{<:Real}, k::Int)
```

Reply:
Generally, the model would be more flexible if `X` is not an argument but the model is `condition`ed on `X` if it is observed.

Reply:
True - this solution makes the model less restrictive. Following changes are made:
1. using ```@model function pPCA(X::AbstractMatrix{<:Real}, k::Int)```
2. Keeping `X` as an argument (conditioning on the obervations). A further question: if `X` is not an argument for model input, it would be a global constant variable, is it correct?